unimod_mapper/unimod_mapper.py

```diff
@@ -498,6 +517,64 @@ def _map_key_2_index_2_value(self, map_key, return_key):
     def _data_list_2_value(self, index, return_key):
         return self.data_list[index][return_key]
 
+    def writeXML(self, modification_dict, xml_file=None):
+        """
+        Writes a unimod-style userdefined_unimod.xml file in
+        at the same location as the unimod.xml
+
+        Args:
+            modification_dict (dict): dictionary containing at least
+            'mass' (mass of the modification),
+            'name' (name of the modificaton),
+            'composition' (chemical composition of the modification as a dictionary {element:number})
+        """
+        if xml_file == None:
+            xml_file = Path(__file__).parent / "userdefined_unimod.xml"
+        unimod = ET.Element("{usermod}unimod")
+        modifications = ET.SubElement(unimod, "{usermod}modifications")
+        mod_dicts = [modification_dict]
+        if os.path.exists(xml_file):
+            data_list = self._parseXML(xml_file=xml_file)
+            for data_dict in data_list:
+                mod_dict = {
+                    "mass": data_dict["mono_mass"],
+                    "name": data_dict["unimodname"],
+                    "composition": data_dict["element"],
+                    "id": data_dict["unimodID"],
+                }
+                mod_dicts.insert(-1, mod_dict)
+
+        for modification_dict in mod_dicts:
+            if modification_dict.get("id", None) == None:
+                modification_dict["id"] = "u{0}".format(len(mod_dicts))
+            mod = ET.SubElement(
+                modifications,
+                "{usermod}mod",
+                title=modification_dict["name"],
+                record_id=modification_dict["id"],
+            )
+            delta = ET.SubElement(
+                mod, "{usermod}delta", mono_mass=str(modification_dict["mass"])
+            )
+
+            for symbol, number in modification_dict["composition"].items():
+                element = ET.SubElement(
+                    delta, "{usermod}element", symbol=symbol, number=str(number)
+                )
+
+        tree = ET.ElementTree(unimod)
+        tree.write(xml_file, encoding="utf-8")
+        xml = xmldom.parse(xml_file)
+        pretty_xml_as_string = xml.toprettyxml()
+        with open(xml_file, "w") as outfile:
+            print(pretty_xml_as_string, file=outfile)
+        self._reparseXML()
```

Comment:
see my comment above. Basically, if you defined `xml_file` it will not be considered by `_reparseXML()`